app/lib/actions/katello/host/upload_package_profile.rb

```diff
@@ -39,7 +39,7 @@ def self.upload(host_id, profile)
             Rails.logger.warn("Host with ID %s has no content facet, continuing" % host_id)
           else
             begin
-              unless SmartProxy.pulp_primary&.pulp3_repository_type_support?(::Katello::Repository::YUM_TYPE)
+              if SmartProxy.pulp_primary&.has_feature?(SmartProxy::PULP_FEATURE)
```

Comment:
Shouldn't it check if yum is actually handled by Pulp 2?

Reply:
Pulp 2 doesn't support reporting specific content types/plugins, i supposed we could check that yum is enabled for katello and pulp2 is enabled, but i don't know that running with yum disabled is actually supported.

Reply:
Doesn't it still make sense to check `pulp2_preferred_for_type?` for yum? Maybe these questions are more for my own understanding of flows.

Reply:
the problem is that even if pulp3 is the preferred type, if you are using pulp2 for katello-agent actions (errata install), pulp2 requires a profile to be uploaded.  This is likely a bug in pulp2 from long ago when pulp actually needed this information to install an errata, but now it really doesn't (but thats moot at this point).  So this needs to continue to upload a package profile to pulp if katello-agent is planning to be used with pulp2.

Reply:
Thanks, that makes sense.